public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -365,6 +366,59 @@ export function rulerRuleToFormValues(ruleWithLocation: RuleWithLocation): RuleF
   }
 }
 
+export function grafanaRuleDtoToFormValues(rule: RulerGrafanaRuleDTO, namespace: string): RuleFormValues {
+  const defaultFormValues = getDefaultFormValues();
+
+  const ga = rule.grafana_alert;
+  const duration = rule.for;
+  const annotations = rule.annotations;
+  const labels = rule.labels;
+  if (rulerRuleType.grafana.rule(rule)) {
```

Comment:
According to the comment, it should handle recording rules, so shouldn't the condition be `rulerRuleType.grafana.recordingRule(rule)`?

Reply:
opsss